controllers/k8ssandra/datacenters.go

```diff
@@ -90,6 +101,22 @@ func (r *K8ssandraClusterReconciler) reconcileDatacenters(ctx context.Context, k
 
 		if err = remoteClient.Get(ctx, dcKey, actualDc); err == nil {
 			// cassdc already exists, we'll update it
+
+			// We need to reevaluate rebuildNeeded again. After the CassandraDatacenter is
+			// created, rebuildNeeded will be false on a subsequent reconciliation when we hit this
+			// point, so we have to check for the presence of the rebuild label on the actual
+			// CassandraDatacenter.
+			if _, rebuildNeeded = actualDc.Labels[api.RebuildLabel]; rebuildNeeded {
+				desiredDc.Labels[api.RebuildLabel] = "true"
+				// We need to recompute and reset the resource annotation here. On the
+				// first reconciliation after a DC has been added rebuildNeeded will be
+				// true and included in the resource hash. On subsequent reconciliations
+				// where the CassandraDatacenter exists rebuildNeeded will initially be set
+				// to false. If we get here, the label is present and thus the resource hash
+				// needs to be updated.
+				annotations.AddHashAnnotation(desiredDc)
+			}
+
 			if err = r.setStatusForDatacenter(kc, actualDc); err != nil {
 				logger.Error(err, "Failed to update status for datacenter")
 				return result.Error(err), actualDcs
```

Comment:
>A question about line 130 below: you are changing `desiredDc` but you aren't recomputing the hash annotation, is that ok?

I assume you are referring to this:

```go
if actualDc.Spec.SuperuserSecretName != desiredDc.Spec.SuperuserSecretName {
		// If actualDc is created with SuperuserSecretName, it can't be changed anymore. We should reject all changes coming from K8ssandraCluster
		desiredDc.Spec.SuperuserSecretName = actualDc.Spec.SuperuserSecretName
		err = fmt.Errorf("tried to update superuserSecretName in K8ssandraCluster")
		logger.Error(err, "SuperuserSecretName is immutable, reverting to existing value in CassandraDatacenter")
}
```
That's actually in main already. I agree that we should enforce that the field remains immutable.

Reply:
Yeah sorry, it's outside the scope of this PR, we can talk about this later.